File: perf.py
from contextlib import contextmanager
from time import time
from sys import stdout
from sklearn.feature_extraction.text import CountVectorizer, TfidfTransformer
from sklearn.feature_selection import SelectKBest, chi2
from sklearn.cross_validation import train_test_split
from sklearn.ensemble import RandomForestClassifier
import os
import numpy as np
import pandas as pd
@contextmanager

# A timer function
def duration(outfile=stdout):
    start = time()
    yield
    end = time()
    outfile.write(str(end - start))

PATH = "/Users/jfrolich/Documents/Data sets/Reuters/fetch/export"
metadata = pd.read_table(os.path.join(PATH, 'metadata.csv'), sep=',')
y = np.array(metadata.acq)

stdout.write('Python,x,')

with duration():
    content = [open(os.path.join(PATH, 'text', str(f))).read() for f in metadata.id]

stdout.write(',')

with duration():
    vectorizer = CountVectorizer(decode_error = 'ignore', ngram_range = (1,3))
    X = vectorizer.fit_transform(content)
    feature_names = vectorizer.get_feature_names()

stdout.write(',')
with duration():
  select = SelectKBest(chi2, k=1000)
  X = select.fit_transform(X, y)

stdout.write('\n')

File: perf.jl
include("text_processing.jl")
using TextProcessing
using DataFrames
using Datetime


PATH = "/Users/jfrolich/Documents/Data sets/Reuters/fetch/export"
metadata = readtable(string(PATH, "/", "metadata.csv"))
y = array(metadata[:acq])
files = [string(PATH, "/text/", id) for id in (metadata[:id])]
# dc = text_array(text_from_files(files))

# println(@elapsed X, feature_names = dtm(dc,3))

# timing_1 = @elapsed dc = text_files(files)
timing_1 = @elapsed dc = text_array(text_from_files(files))
timing_2 = @elapsed X, feature_names = dtm(dc,3)
timing_3 = @elapsed sel_features = select_k_best(X, y, chi2, 10000)

println("Julia,", now(), ",", timing_1, ",", timing_2, ",", timing_3)
